File: 03 Berechnungen/Pascal/Aufgabe8.pas
Program Aufgabe8;

{$codepage utf8}

Uses sysutils;

Var Zahl1: integer = 1;
    Zahl2: integer;
    Kommazahl1: real = 3.14;
    Kommazahl2: real;
    String1: String = '2';
    String2: String = '2.71';
    String3: String;
    String4: String;

Begin
    WriteLn('Zahl1 = ', Zahl1, ', Kommazahl1 = ', Kommazahl1:1:2, ', String1 = "', String1, ', String2 = "', String2, '"');

    String3 := IntToStr(Zahl1);
    WriteLn('String3 = IntToStr(Zahl1); String3 = "', String3, '"');

    String4 := FloatToStr(Kommazahl1);
    WriteLn('String4 = FloatToStr(Kommazahl1); String4 = "', String4, '"');

    Zahl2 := StrToInt(String1);
    WriteLn('Zahl2 = StrToInt(String1); Zahl2 = ', Zahl2);

    Kommazahl2 := StrToFloat(String2);
    WriteLn('Kommazahl2 = StrToFloat(String2); Kommazahl2 = ', Kommazahl2:1:2);
End.

File: 03 Berechnungen/Go/Aufgabe8.go
package main

import (
	"fmt"
	"strconv"
)

var (
	zahl1      int = 1
	zahl2      int
	kommazahl1 float64 = 3.14
	kommazahl2 float64
	string1    string = "2"
	string2    string = "2.71"
	string3    string
	string4    string
)

func main() {
	fmt.Println("zahl1 =", zahl1, ", kommazahl1 =", kommazahl1, ", string1 = \"", string1, "\", string2 = \"", string2, "\"")

	fmt.Println()

	string3 = strconv.Itoa(zahl1)
	fmt.Println("string3 = strconv.Itoa(zahl1); string3 = \"", string3, "\"")

	string4 = strconv.FormatFloat(float64(kommazahl1), 'f', 2, 32)
	fmt.Println("string4 = strconv.FormatFloat(float64(kommazahl1), 'f', 2, 32); string4 = \"", string4, "\"")

	zahl2, _ = strconv.Atoi(string1)
	fmt.Println("zahl2 = strconv.Atoi(string1); zahl2 =", zahl2)

	kommazahl2, _ = strconv.ParseFloat(string2, 32)
	fmt.Println("kommazahl2 = strconv.ParseFloat(string2, 32); kommazahl2 =", kommazahl2)
}
